src/Servers/Kestrel/Kestrel/src/WebHostBuilderKestrelExtensions.cs

```diff
@@ -32,6 +33,7 @@ public static IWebHostBuilder UseKestrel(this IWebHostBuilder hostBuilder)
                 services.TryAddSingleton<ITransportFactory, SocketTransportFactory>();
 
                 services.AddTransient<IConfigureOptions<KestrelServerOptions>, KestrelServerOptionsSetup>();
+                services.AddSingleton<IHttpContextFactory, KestrelHttpContextFactory>();
```

Comment:
So what happens when IIS in-proc implements the same feature?

Reply:
What are you asking? When happens if somebody replaces this feature? Kestrel will work but will be un-optimized.

Reply:
I can test the performance of the interface approach and make changes to the default factory. That would be a bit more cooperative.